src/core/stream/orchestrator/stream_orchestrator.ts

```diff
@@ -172,11 +175,18 @@ export default function StreamOrchestrator(
       return EVENTS.activePeriodChanged(period);
     }));
 
+  const isLastPeriodKnown$ = fromEvent(manifest, "manifestUpdate").pipe(
+    map(() => manifest.isLastPeriodKnown),
+    startWith(manifest.isLastPeriodKnown),
+    distinctUntilChanged()
+  );
+
   // Emits an "end-of-stream" event once every PeriodStream are complete.
   // Emits a 'resume-stream" when it's not
-  const endOfStream$ = areStreamsComplete(...streamsArray)
-    .pipe(map((areComplete) =>
-      areComplete ? EVENTS.endOfStream() : EVENTS.resumeStream()
+  const endOfStream$ = combineLatest([areStreamsComplete(...streamsArray),
+                                      isLastPeriodKnown$])
+    .pipe(map(([areComplete, isLastPeriodKnown]) =>
+      (areComplete && isLastPeriodKnown) ? EVENTS.endOfStream() : EVENTS.resumeStream()
```

Comment:
Shouldn't a `distinctUntilChanged` also be present here?
Even if that may be impossible for now to send two endOfStream/resumeStream in a row, I don't see it as particularly evident from just considering the Observables that we listen to.